### DISCUSSION TITLE
Make one of the domain as inactive for solving the equation

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#19674] [Q&A Modules: General] [answered]
Hello MOOSE experts

I am solving chemo-mechanical equations using kernels which I modified as per my requirement. 

Let's say there are two domains in the mesh file. And I want to solve the diffusion equation for only one of the domains. However, the mechanical equation should be solved for both of them.

Can you please suggest some examples or procedures do it?

Thank you very much in advance.

Best
Avtar

**Comment by GiudGiud** [comment, marked as answer]
Hello

yes you can specify a `block` argument on the kernels and/or on the relevant variables to restrict some physics to parts of the domain.
Here are a few inputs that do it:
https://github.com/idaholab/moose/blob/next/test/tests/kernels/block_kernel/block_vars.i. (restricts via the variable relevant to a kernel)
https://github.com/idaholab/moose/blob/next/test/tests/kernels/block_kernel/block_kernel_test.i (restricts the body force kernel to a single block)

Guillaume

**Reply by avtarsinghh1991** [reply]
Hello Guillaume

Thank you very much. I am using **block** argument to specify the active and inactive domains. It's working fine. But there is a small issue.

I am simulating particle embedded in a matrix with only particle treated as an active domain for diffusion.
I am providing flux on the surface of the particle.
![Screenshot 2022-01-03 142613](https://user-images.githubusercontent.com/94848840/147971443-b9e7c663-713f-4bb1-966d-f069d654dd85.png)

Diffusion works fine within the particle i.e., it increases with time. But -ve value comes right above the particle surface whose value increases with time. I have a figure here for your reference.

I have tried with different mesh sizes but it keeps on popping out for every case with value increasing with time.

Can you please help me to remove that scenario? I shall be very thankful.

Best
Avtar

**Reply by GiudGiud** [reply]
Hello

What is this `diffused` field? Is it a postprocessing of the flux? What is defining it?
Should it be defined outside the particle if only the particle is active for diffusion?

Guillaume

**Reply by avtarsinghh1991** [reply]
Hello Guillaume

I named the concentration field as "diffused". I am solving the concentration field only for particle by applying the flux boundary condition on its surface.

It means that concentration field outside the particle should remain zero. But the concentration right above the particle surface begins to decrease and keeps on decreasing with time.


**Input file: There are 33 blocks with last one being the outer domain.**

[Kernels]
  [diffusion]
    type = Concentration 
    variable = diffused
 block = '0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31' **# 32 is outside domain so not considered**
  [../]
  [./euler]
    type = DiffusionTimeDerivative
    variable = diffused
  [../]
[]

[Materials]
[./diffusivity_tensor_with_Euler]
 type = AnisoDiffusivity
 read_prop_user_object = euler_angle_read
 diffusivity = '7.08e-15 0 0 0 10.08e-15 0 0 0 4.08e-15'
 block = '0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31'
[../]
[]

This may be due to interpolation of concentration field across 1-2 elements.


Best
Avtar

**Reply by GiudGiud** [reply]
Is there any kernel acting on the concentration field on the outside of the particle (where it goes negative right?)

**Reply by avtarsinghh1991** [reply]
Yes, TensorMechanics kernel is active for both the domains.

[Modules/TensorMechanics/Master]
  [all]
    add_variables = true
    strain = FINITE
    automatic_eigenstrain_names = true
    generate_output = 'vonmises_stress'
  []
[]

**Reply by GiudGiud** [reply]
But TensorMechanics does not act on diffused right?
If diffused is supposed to be constant outside the particle, I would block restrict the variable to the particle, on top of the block restriction you have for the kernels

**Reply by avtarsinghh1991** [reply]
I am solving the coupled chemo-mechanical problem. So, diffusion causes stress but stress is not affecting the diffusion.

So, the system is only one way coupled.

Yes, concentration should be solved only within the particle.

Update: One more thing, the region in which concentration becomes negative outside the particle is always spread upto two mesh elements along radial direction even if I decrease the size of the meshing.

**Reply by GiudGiud** [reply]
Let's try restricting the variable then.
What FE family is diffused btw?

**Reply by avtarsinghh1991** [reply]
Thanks

Sure. I am trying to restrict the variable. Will let you know the output.

All variables belong to lagrange family.

[Variables]
  [./diffused]
    initial_condition = 0.0
    order = FIRST
    family = LAGRANGE
  [../]
[./disp_x]
   initial_condition = 0.0
    order = FIRST
    family = LAGRANGE
[../]
[./disp_y]
   initial_condition = 0.0
    order = FIRST
    family = LAGRANGE
[../]
[]

**Reply by avtarsinghh1991** [reply]
Hello Guillaume

Thank you very much. The problem has been resolved.

The issue was, I was not putting the block items in the time derivative module.
```
[Kernels]
[diffusion]
type = Concentration
variable = diffused
block = '0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31' # 32 is outside domain so not considered
[../]
[./euler]
type = TimeDerivative
variable = diffused
### block = '0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31' # 32 is outside domain so not considered
[../]
[]
```
Best
Avtar
